cob_phidget_em_state/scripts/em_state_phidget.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+
+import rospy
+from cob_phidgets.msg import AnalogSensor
+from cob_phidgets.msg import DigitalSensor
+from cob_msgs.msg import PowerState
+from cob_msgs.msg import EmergencyStopState
+
+class EmState():
+    ST_EM_FREE = 0
+    ST_EM_ACTIVE = 1
+    ST_EM_CONFIRMED = 2
+
+class EMStatePhidget():
+    def __init__(self):
+        self.last_update = rospy.Time(0)
+        self.em_caused_by_button = False
+
+        self.last_rear_em_state = False
+        self.last_front_em_state = False
+
+        self.em_stop_status = EmState.ST_EM_ACTIVE
+        self.em_msg = EmergencyStopState()
+
+        self.sub_digital_sensors = rospy.Subscriber("digital_sensors", DigitalSensor, self.phidget_cb)
+        self.pub_em_state = rospy.Publisher('em_stop_state', EmergencyStopState, queue_size=1)
+
+    def phidget_cb(self, msg):
+        self.last_update = rospy.Time.now()
+        front_em_active = False
+        rear_em_active = False
+        got_message = False
+
+        for i in range(0, len(msg.uri)):
```

Comment:
oh and just one remark on writing code comment. Please set your comment below the code block and not above. This way the code lines are also visible on the conversation page.